util/roomValidate.js

```diff
@@ -0,0 +1,43 @@
+module.exports = function (rooms, config) {
+    return {
+        username: function (username, room, callback) {
+            var error;
+            if (typeof username === 'undefined'
+                || username === ''
+                || username === null
+                || username.match(/server/i)) {
+                error = {
+                    name: 'adduserfail',
+                    arg: 'Invalid username: ' + username
+                };
+                return callback(error, false);
+            } else if (rooms[room].usernames.indexOf(username) !== -1) {
+                error = {
+                    name: 'adduserfail',
+                    arg: 'Invalid username, already in use: ' + username
+                };
+                return callback(error, false);
+            } else if (username.length > config.maxUserNameLength) {
+                error = {
+                    name: 'adduserfail',
```

Comment:
With regards to `name: 'addUserFail'`: My first instinct is that we shouldn't set/change the data on an erroneous operations.  Like maybe arg is fine or we return some object that wraps the return value and it has the data, a status, and a more detailed message like if an error occurred it would say what arg says and status would be like > 0 or "error" or something like that maybe.  That might be to heavy handed for a relatively simple function but I'm not sure I'm on board with setting name to addUserFail.